R/shiny_module.R

```diff
@@ -0,0 +1,219 @@
+#' Profvis UI for Shiny Apps
+#'
+#' Provides the UI for the Profvis module in Shiny apps. This module creates a
+#' profvis interface to profile parts of a shiny app and save the profvis or
+#' Rprof output. Used with \code{\link{profvis_server}}. This function requires
+#' the \code{shiny} package.
+#'
+#' @param id Output id from \code{profvis_server}.
+#'
+#' @examples
+#' if(interactive()) {
+#'   shinyApp(
+#'     fluidPage(
+#'       plotOutput("plot"),
+#'       actionButton("new", "New plot"),
+#'       profvis_ui("profiler")
+#'     ),
+#'     function(input, output, session) {
+#'       callModule(profvis_server, "profiler")
+#'
+#'       output$plot <- renderPlot({
+#'         input$new
+#'         ggplot(diamonds, aes(carat, price)) + geom_point()
+#'       })
+#'     }
+#'   )
+#' }
+#'
+#' @export
+profvis_ui <- function(id) {
+  if (!requireNamespace("shiny", quietly = TRUE)) {
+    stop('profvis_ui requires the shiny package.')
+  }
+  ns <- shiny::NS(id)
+
+  style <- htmltools::css(
+    padding = "6px",
+    white_space = "nowrap",
+    top = "-1px",
+    border_top_left_radius = "0",
+    border_top_right_radius = "0",
+    box_shadow = "none",
+    z_index = 9000
+  )
+
+  shiny::fixedPanel(
+    top = 0, left = -200, width = "auto", height = "auto",
+    class = "profvis-module-container well", style = style, draggable = TRUE,
+
+    shiny::uiOutput(ns("button_group"), class = "btn-group")
+  )
+}
+
+#' Profvis Server for Shiny Apps
+#'
+#' Provides the server for the Profvis module in Shiny apps. This module creates
+#' a profvis interface to profile parts of a shiny app and save the profvis or
+#' Rprof output. Used with \code{\link{profvis_ui}}. This function requires the
+#' \code{shiny} package.
+#'
+#' @param input,output,session Arguments used by
+#'   \code{\link[shiny]{callModule}}.
+#' @param dir Output directory to save Rprof files.
+#'
+#' @examples
+#' if(interactive()) {
```

Comment:
If this example is the exact same as `profvis_ui`, could call `example("profvis_ui", "profvis")`

Reply:
(^^ Style choice)